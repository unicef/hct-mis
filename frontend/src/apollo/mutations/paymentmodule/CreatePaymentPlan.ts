import { gql } from 'apollo-boost';

export const CREATE_PAYMENT_PLAN = gql`
  mutation CreatePP($input: CreatePaymentPlanInput!) {
    createPaymentPlan(input: $input) {
      paymentPlan {
        id
      }
    }
  }
`;
