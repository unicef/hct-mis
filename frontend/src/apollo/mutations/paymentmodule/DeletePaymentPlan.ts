import { gql } from 'apollo-boost';

export const DELETE_PAYMENT_PLAN = gql`
  mutation DeletePP($paymentPlanId: ID!) {
    deletePaymentPlan(paymentPlanId: $paymentPlanId) {
      paymentPlan {
        id
        status
      }
    }
  }
`;
