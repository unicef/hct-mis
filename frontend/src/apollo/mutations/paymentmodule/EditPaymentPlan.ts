import { gql } from 'apollo-boost';

export const UPDATE_PAYMENT_PLAN = gql`
  mutation UpdatePP($input: UpdatePaymentPlanInput!) {
    updatePaymentPlan(input: $input) {
      paymentPlan {
        id
      }
    }
  }
`;
