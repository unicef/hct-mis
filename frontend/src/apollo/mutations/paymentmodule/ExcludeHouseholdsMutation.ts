import { gql } from 'apollo-boost';

export const EXCLUDE_HOUSEHOLDS_PP = gql`
  mutation ExcludeHouseholdsPP($input: //TODO!!!) {
    excludeHouseholds(input: $input) {
      paymentPlan {
        id
        excludedHouseholdsIds
    }
  }
`;
