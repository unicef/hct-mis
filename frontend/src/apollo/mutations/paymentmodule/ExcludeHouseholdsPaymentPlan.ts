import { gql } from 'apollo-boost';

export const EXCLUDE_HOUSEHOLDS_PP = gql`
  mutation ExcludeHouseholdsPP(
    $paymentPlanId: ID!
    $excludedHouseholdsIds: [String]!
  ) {
    excludeHouseholds(
      paymentPlanId: $paymentPlanId
      excludedHouseholdsIds: $excludedHouseholdsIds
    ) {
      paymentPlan {
        id
        excludedHouseholds {
          id
          unicefId
        }
      }
    }
  }
`;
