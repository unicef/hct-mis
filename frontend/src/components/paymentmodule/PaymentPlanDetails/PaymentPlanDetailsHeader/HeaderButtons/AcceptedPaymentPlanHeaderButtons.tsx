import { Box, Button } from '@material-ui/core';
import React, { useState } from 'react';
import { useTranslation } from 'react-i18next';
import styled from 'styled-components';
import { PaymentPlanQuery } from '../../../../../__generated__/graphql';

const ButtonContainer = styled.span`
  margin: 0 ${({ theme }) => theme.spacing(2)}px;
`;

export interface AcceptedPaymentPlanHeaderButtonsProps {
  paymentPlan: PaymentPlanQuery['paymentPlan'];
  canDownloadXlsx: boolean;
  canSendToFsp: boolean;
}

export const AcceptedPaymentPlanHeaderButtons = ({
  paymentPlan,
  canDownloadXlsx,
  canSendToFsp,
}: AcceptedPaymentPlanHeaderButtonsProps): React.ReactElement => {
  const { t } = useTranslation();
  const [openApprove, setOpenApprove] = useState(false);
  const [openDuplicate, setOpenDuplicate] = useState(false);
  const [openDelete, setOpenDelete] = useState(false);
  return (
    <Box display='flex' alignItems='center'>
      {canDownloadXlsx && (
        <ButtonContainer>
          <Button
            variant='contained'
            color='primary'
            onClick={() => setOpenApprove(true)}
          >
            {t('Download XLSX')}
          </Button>
        </ButtonContainer>
      )}
      {canSendToFsp && (
        <ButtonContainer>
          <Button
            variant='contained'
            color='primary'
            onClick={() => setOpenApprove(true)}
          >
            {t('Send to Fsp')}
          </Button>
        </ButtonContainer>
      )}
    </Box>
  );
};
