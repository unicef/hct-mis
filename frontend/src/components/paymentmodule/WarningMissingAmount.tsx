import { Box } from '@material-ui/core';
import WarningIcon from '@material-ui/icons/Warning';
import React from 'react';
import { useTranslation } from 'react-i18next';
import styled from 'styled-components';

const WarnIcon = styled(WarningIcon)`
  color: ${({ theme }) => theme.hctPalette.orange};
`;

interface WarningMissingAmountProps {
  amount: number;
  currency: string;
}

export function WarningMissingAmount({
  amount,
  currency,
}: WarningMissingAmountProps): React.ReactElement {
  const { t } = useTranslation();

  return (
    <Box display='flex' alignItems='center'>
      <WarnIcon />
      {t('Missing')} {amount} {currency}
    </Box>
  );
}
