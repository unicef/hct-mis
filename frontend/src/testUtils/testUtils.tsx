import React, { ReactElement } from 'react';
import { BrowserRouter } from 'react-router-dom';
import { render, RenderOptions } from '@testing-library/react';
import cloneDeep from 'lodash/cloneDeep';
import { TestProviders } from './testProviders';

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'queries'>,
) =>
  render(<BrowserRouter>{ui}</BrowserRouter>, {
    wrapper: TestProviders,
    ...options,
  });

export * from '@testing-library/react';
export { customRender as render };

let m_w = 123456789;
let m_z = 987654321;
const mask = 0xffffffff;

// Takes any integer
export const seed = (i) => {
  m_w = (123456789 + i) & mask;
  m_z = (987654321 - i) & mask;
};

// Returns number between 0 (inclusive) and 1.0 (exclusive),
// just like Math.random().
export const random = () => {
  m_z = (36969 * (m_z & 65535) + (m_z >> 16)) & mask;
  m_w = (18000 * (m_w & 65535) + (m_w >> 16)) & mask;
  let result = ((m_z << 16) + (m_w & 65535)) >>> 0;
  result /= 4294967296;
  return result;
};

export const makeApolloLoadingMock = (mocks) => {
  let mocksWithDelay = cloneDeep(mocks);
  for (const mock of mocksWithDelay) {
    mock.request.delay = 1e21;
  }
  return mocksWithDelay;
};
