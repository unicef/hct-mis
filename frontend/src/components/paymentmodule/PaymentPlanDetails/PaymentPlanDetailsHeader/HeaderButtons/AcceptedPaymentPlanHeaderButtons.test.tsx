import { MockedProvider } from '@apollo/react-testing';
import { act } from 'react-dom/test-utils';
import wait from 'waait';
import React from 'react';
import { fakeApolloPaymentPlan } from '../../../../../../fixtures/paymentmodule/fakeApolloPaymentPlan';
import { fakeExportXlsxPpListPerFspMutation } from '../../../../../../fixtures/paymentmodule/fakeExportXlsxPpListPerFspMutation';
import { render } from '../../../../../testUtils/testUtils';
import { AcceptedPaymentPlanHeaderButtons } from './AcceptedPaymentPlanHeaderButtons';

describe('components/paymentmodule/PaymentPlanDetails/PaymentPlanDetailsHeader/HeaderButtons/AcceptedPaymentPlanHeaderButtons', () => {
  it('should render with no buttons', async () => {
    const { container } = render(
      <MockedProvider
        addTypename={false}
        mocks={fakeExportXlsxPpListPerFspMutation}
      >
        <AcceptedPaymentPlanHeaderButtons
          canDownloadXlsx={false}
          canSendToFsp={false}
          paymentPlan={fakeApolloPaymentPlan}
        />
      </MockedProvider>,
    );
    await act(() => wait(0)); // wait for response

    const buttons = container.querySelectorAll('button');
    expect(buttons).toHaveLength(0);

    expect(container).toMatchSnapshot();
  });

  it('should render with buttons', async () => {
    const { container } = render(
      <MockedProvider
        addTypename={false}
        mocks={fakeExportXlsxPpListPerFspMutation}
      >
        <AcceptedPaymentPlanHeaderButtons
          canDownloadXlsx
          canSendToFsp
          paymentPlan={fakeApolloPaymentPlan}
        />
      </MockedProvider>,
    );
    await act(() => wait(0)); // wait for response

    const buttons = container.querySelectorAll('button');
    expect(buttons).toHaveLength(2);

    expect(container).toMatchSnapshot();
  });
});
