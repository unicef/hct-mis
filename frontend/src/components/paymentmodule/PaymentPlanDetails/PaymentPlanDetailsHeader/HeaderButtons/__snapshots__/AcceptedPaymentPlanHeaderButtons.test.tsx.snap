// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/paymentmodule/PaymentPlanDetails/PaymentPlanDetailsHeader/HeaderButtons/AcceptedPaymentPlanHeaderButtons should render with buttons 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-2"
  >
    <div
      class="MuiBox-root MuiBox-root-3"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary Mui-disabled Mui-disabled"
        data-cy="button-export-xlsx"
        disabled=""
        tabindex="-1"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          <span
            class="MuiButton-startIcon MuiButton-iconSizeMedium"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 9h-4V3H9v6H5l7 7 7-7zM5 18v2h14v-2H5z"
              />
            </svg>
          </span>
          Export Xlsx
        </span>
      </button>
    </div>
    <div
      class="MuiBox-root MuiBox-root-4"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Send to FSP
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`components/paymentmodule/PaymentPlanDetails/PaymentPlanDetailsHeader/HeaderButtons/AcceptedPaymentPlanHeaderButtons should render with no buttons 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-1"
  />
</div>
`;
